bokeh/core/validation/check.py

```diff
@@ -125,10 +125,6 @@ def check_integrity(models):
         if code not in __silencers__:
             log.warning("W-%d (%s): %s: %s" % msg)
```

Comment:
All the `log` calls in this function will have to be removed. Instead, the function should  collect all the warnings and errors into some data structure, and then return that data structure.